infra/base-images/base-builder/compile

```diff
@@ -22,13 +22,13 @@ if [ -z "${SANITIZER_FLAGS-}" ]; then
   export SANITIZER_FLAGS=${!FLAGS_VAR-}
 fi
 
-if [[ $FUZZING_ENGINE != "none" ]]; then
-  # compile script might override environment, use . to call it.
-  . compile_${FUZZING_ENGINE}
-fi
+if [[ $SANITIZER_FLAGS = *sanitize=memory* ]]; then
+  if [[ $FUZZING_ENGINE != "none" ]]; then
```

Comment:
Don't like repeating this if [[ $FUZZING_ENGINE != "none" ]];, can we set a FUZZING_ENGINE_SANITIZER_FLAGS var inside this sanitizer=memory

and then
do later like 
SANITIZER_FLAGS="$FUZZING_ENGINE_SANITIZER_FLAGS" . compile_${FUZZING_ENGINE}
?
thoughts ?

Reply:
Sounds good, let me do it that way.